theano/__init__.py

```diff
@@ -24,6 +24,16 @@
 
 __docformat__ = "restructuredtext en"
 
+# Check if theano is being imported from a child process
+# If yes, raises error, refer issue #1064
+import os
+if 'THEANO_MAX_GPU' in os.environ:
```

Comment:
This would need a test in theano/tests/test_init.py

In that test, make a sub process that import theano and make sure it raise an error. It should suppose theano tests was started without that env variable.